### ISSUE TITLE
Bevy - Server shutdown?

### ISSUE THREAD

**Issue by brandon-reinhart** [naia-lib/naia#92]
It doesn't appear that there's a way to shutdown or stop listening to a port once it's started? The server API has listen, but no way to close. Dropping the server resource doesn't seem to release the socket/port. What's the expected way to shutdown a bevy server resource?

**Comment by brandon-reinhart** [comment]
Looks like socket::listen detaches the executors but doesn't hold onto the channels. So if you want to close a socket, you'd want to close the channels. Then the executor should check if the channels are closed (or recv is in error re: closed channels) and end the loop and die. The channels could be closed externally, thereby resulting in a shutdown. 

Shutdown is async, so there'd also be a need to check that the shutdown is complete and the port is available before rebinding it at some point later...

Is this a correct reading?

**Comment by connorcarpenter** [comment]
Yep this feature is definitely needed, there's no way to cleanly shut down a server and clean up it's listening ports / resources. I've been relying on the OS to do that after stopping the running process. Thanks again for this :pray:
